### PR TITLE
Add support for disabling hardware keys.

### PR DESCRIPTION
**Pull request by Nihhaar**
* Import changes from ferrari-dev-team

### CODE REVIEW

File: drivers/input/touchscreen/atmel_mxt_ts_336t.c [hunk 1]
```diff
@@ -1573,6 +1574,11 @@ static void mxt_proc_t97_messages(struct mxt_data *data, u8 *msg)
 	if (data->input_dev == NULL)
 		return;
 
+	if(data->disable_keys) {
+		dev_err(&data->client->dev, "keyarray is disabled\n");
```
Comment (thewisenerd) [inline drivers/input/touchscreen/atmel_mxt_ts_336t.c]: dev_dbg this please. don't want it spamming kmsg.

**Comment by thewisenerd** [comment]
5bf08cb3b7 is not needed.

there's no reason to merge a build script in. don't want it causing any confusions.

**Comment by thewisenerd** [comment]
you _might_ also wanna consider letting the disable_keys patch's author remain.

**Comment by thewisenerd** [comment]
works fine. merging, thanks :]

**Comment by Nihhaar** [comment]
:) :D

**Pull request merged**